### PR TITLE
Removing CartesionProductExtension version 1 (#517 / #635)

### LINKED ISSUE #517: CartesianTest clean-up
After merging #487, code duplication went over 3%, which was part of [Pioneer's quality gate](https://sonarcloud.io/organizations/junit-pioneer-xp/quality_gates/show/39036). In order to get back a successful build on `main` (and elsewhere), we increased that metric to 7%.

What's left to do:

* Remove duplication
* Reset quality gate
* Publish Pioneer v2.0

We aim to do that in December 2021.

### PR DESCRIPTION
**Pull request by Bukama**
Closes #517.

```
Remove CartesianTest extension v1 (#517 / #635)

The CartesianTest extension was reimplemented in #487.
At the same time v1 was marked as deprecated and for
removal in Pioneer 2.0.

This PR removes this deprecated version.

Closes: #517
PR: #635
```

### CODE REVIEW

**Review by nipafx — COMMENTED** [review]
I can't wait to kick this out, but I think in the end it should be @Michael1993 that merges this PR - it was him who wrote most of that code.

I don't want to remove `cartesian-product-v1.adoc`. It would break a URL (not many people know this, but every time that happens Tim Berners-Lee drowns a kitten) and remove documentation that users of JUnit Pioneer 1.x may rely on. Instead, replace the note in lines 5-8 with this:

```adoc
NOTE: The CartesianTest extension has undergone significant changes.
This page covers the old version (in package `org.junitpioneer.jupiter`), which was deprecated in 1.6.0 and removed in 2.0.
The new version (in package `org.junitpioneer.jupiter.cartesian`) has link:/docs/cartesian-product[its own documentation page]. 
```

And can you replace the note in `cartesian-product.adoc` (lines 21-23) with this:

```adoc
NOTE: This page documents the second version of the CartesianTest extension.
The first version was deprecated 1.6.0 and removed in 2.0 - link:/docs/cartesian-product-v1[find its documentation here].
```

File: src/main/java/org/junitpioneer/jupiter/params/RangeSourceArgumentsProvider.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
  */
 // CartesianAnnotationConsumer is deprecated for removal
 @SuppressWarnings("deprecation")
```
[the remark this answers is not in the document]
Comment (Bukama) [reply src/main/java/org/junitpioneer/jupiter/params/RangeSourceArgumentsProvider.java:47]: 👍

**Comment by Bukama** [comment]
> I don't want to remove `cartesian-product-v1.adoc`. It would break a URL (not many people know this, but every time that happens Tim Berners-Lee drowns a kitten) and remove documentation that users of JUnit Pioneer 1.x may rely on. 

I think there are far too much pets out there, so I don't care about the kitten :D. But I accept your argument of our version 1 users and revoke the removal (also applied your suggested changes)

File: src/main/java/org/junitpioneer/jupiter/params/RangeSourceArgumentsProvider.java [hunk 2]
```diff
@@ -45,8 +45,8 @@
...
-class RangeSourceArgumentsProvider<N extends Number & Comparable<N>> implements ArgumentsProvider,
-		org.junitpioneer.jupiter.CartesianAnnotationConsumer<Annotation>, CartesianParameterArgumentsProvider<N> { //NOSONAR deprecated interface use will be removed in later release
+class RangeSourceArgumentsProvider<N extends Number & Comparable<N>>
+		implements ArgumentsProvider, CartesianParameterArgumentsProvider<N>, java.util.function.Consumer<Annotation> { //NOSONAR deprecated interface use will be removed in later release
```
[the remark this answers is not in the document]
Comment (nipafx) [reply src/main/java/org/junitpioneer/jupiter/params/RangeSourceArgumentsProvider.java]: Will take a look soon.

**Comment by Bukama** [comment]
Blocked until Pioneer 2.0 and the approval of @Michael1993

**Review by beatngu13 — APPROVED** [review]

**Pull request merged**